sktime/forecasting/model_selection/_split.py

```diff
@@ -96,8 +96,7 @@ def _get_end(y_index: pd.Index, fh: ForecastingHorizon) -> int:
     fh_offset = null if fh.is_all_in_sample() else fh[-1]
     if array_is_int(fh):
         return n_timepoints - fh_offset - 1
-    else:
-        return y_index.get_loc(y_index[-1] - fh_offset)
+    return y_index.get_loc(y_index[-1] - fh_offset)
```

Comment:
out of curiosity, why?

Reply:
It is just to make the code more concise, since the `else` clause is actually not needed. Some linters like pylint discourage the use of an `else` after a return statement ([see this](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html)). In any case, it's true that in the end it is probably just a matter of style. I can change it back to the original code if you prefer it.

Reply:
Interesting. I did not know of this linter warning. Still curious what is the deep reason behind it.

Reply:
I have just briefly read some blog posts about this topic ([link](https://blog.mozilla.org/nnethercote/2009/08/31/no-else-after-return-considered-harmful/), [link](https://dmerej.info/blog/post/else-after-return-yea-or-nay/)) and it appears that there is ongoing debate. Ultimately, it seems that the two main points of discussion are readability (which can be quite subjective) and whether it is better to use a functional approach or if it is acceptable to rely on side-effects.

Although I generally prefer the functional style, in this situation I like the conciseness of the approach without else clauses. That being said, after reading those posts, I have come to realize that I may have been following the guidelines of Pylint without questioning the reasoning behind it. If I encounter a similar code pattern in the future, I will likely not make any changes to it as I now understand that it is likely a valid option too.

Reply:
Nice! Thanks! I've learned something too :)

Reply:
if it adds anything - I'm neutral on the issue.

I would consider using `else` as slightly more readable, as it is more verbose - it makes it very explicit that the remaining code is executed only if the condition above is `False`. Whereas otherwise one needs to infer (rather than read) that this is the case, but it's not too hard because understanding `return` is python 101.